#! /bin/sh /usr/share/dpatch/dpatch-run
## 95_fix_hostdb_inserts.dpatch by William Preston
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: The initial inserts of host entries into the objects
## DP: table use the string 'NULL' instead of actual null values.
## DP: backported from git: 4d7721a9c1ccc3a62ceab2445795014c3527dfc0

@DPATCH@
X-Git-Url: https://git.icinga.org/?p=icinga-core.git;a=blobdiff_plain;f=module%2Fidoutils%2Fsrc%2Fdbhandlers.c;h=248abcf93a8e2d83a3e8528ae5f4119f3ab5a423;hp=6bd18d10131f881bffdf0354660111223838b1ee;hb=4d7721a9c1ccc3a62ceab2445795014c3527dfc0;hpb=6198ab4db9f57eb944ce4e56e1651966e908281b;js=1

diff --git a/module/idoutils/src/dbhandlers.c b/module/idoutils/src/dbhandlers.c
index 6bd18d1..248abcf 100644
--- a/module/idoutils/src/dbhandlers.c
+++ b/module/idoutils/src/dbhandlers.c
@@ -401,17 +401,28 @@ int ndo2db_get_object_id_with_insert(ndo2db_idi *idi, int object_type, char *n1,
 		return NDO_OK;
 
 	if (name1 != NULL) {
-		es[0] = ndo2db_db_escape_string(idi, name1);
+		tmp = ndo2db_db_escape_string(idi, name1);
+		asprintf(&es[0], "'%s'", tmp);
+		if (tmp) {
+			free(tmp);
+			tmp = NULL;
+		}
 	} else
 		asprintf(&es[0],"NULL");
+
 	if (name2 != NULL) {
-		es[1] = ndo2db_db_escape_string(idi, name2);
+		tmp = ndo2db_db_escape_string(idi, name2);
+		asprintf(&es[1], "'%s'", tmp);
+		if (tmp) {
+			free(tmp);
+			tmp = NULL;
+		}
 	} else
 		asprintf(&es[1], "NULL");
 
 #ifndef USE_ORACLE /* everything else will be libdbi */
 	if (asprintf(&buf,
-			"INSERT INTO %s (instance_id, objecttype_id, name1, name2) VALUES (%lu, %d, '%s', '%s')",
+			"INSERT INTO %s (instance_id, objecttype_id, name1, name2) VALUES (%lu, %d, %s, %s)",
 			ndo2db_db_tablenames[NDO2DB_DBTABLE_OBJECTS],
 			idi->dbinfo.instance_id, object_type, es[0],
 			es[1]) == -1)
